doltpy/sql/write.py

```diff
@@ -0,0 +1,288 @@
+def write_pandas(engine: Engine,
+                 table: str,
+                 df: pd.DataFrame,
+                 on_duplicate_key_update: bool = True,
+                 create_if_not_exists: bool = False,
+                 primary_key: List[str] = None,
+                 commit: bool = False,
+                 commit_message: str = None,
+                 commit_date: datetime = None,
+                 batch_size: int = DEFAULT_BATCH_SIZE):
+    """
+
+    :param engine:
+    :param table:
+    :param df:
+    :param on_duplicate_key_update:
+    :param create_if_not_exists:
+    :param primary_key:
+    :param commit:
+    :param commit_message:
+    :param commit_date:
+    :param batch_size:
+    :return:
+    """
+    return write_rows(engine,
+                      table,
+                      df.to_dict('records'),
+                      on_duplicate_key_update,
+                      create_if_not_exists,
+                      primary_key, commit,
+                      commit_message,
+                      commit_date,
+                      batch_size)
+
+
+def write_rows(engine: Engine,
+               table_name: str,
+               rows: Iterable[dict],
+               on_duplicate_key_update: bool = True,
+               create_if_not_exists: bool = False,
+               primary_key: List[str] = None,
+               commit: bool = False,
+               commit_message: str = None,
+               commit_date: datetime = None,
+               batch_size: int = DEFAULT_BATCH_SIZE):
+    """
+
+    :param engine:
+    :param table_name:
+    :param rows:
+    :param on_duplicate_key_update:
+    :param create_if_not_exists:
+    :param primary_key:
+    :param commit:
+    :param commit_message:
+    :param commit_date:
+    :param batch_size:
+    :return:
+    """
+    metadata = MetaData(bind=engine)
+    metadata.reflect()
+
+    if table_name not in metadata.tables and create_if_not_exists:
+        infer_table_schema(metadata, table_name, rows, primary_key)
+        metadata.reflect()
+
+    table = metadata.tables[table_name]
+
+    rows = list(rows)
+    for i in range(max(1, math.ceil(len(rows) / batch_size))):
+        batch_start = i * batch_size
+        batch_end = min((i + 1) * batch_size, len(rows))
+        batch = rows[batch_start:batch_end]
+        logger.info(f'Writing records {batch_start} through {batch_end} of {len(rows)} rows to Dolt')
+        write_batch(engine, table, batch, on_duplicate_key_update)
+
+    if commit:
+        commit_tables(engine, commit_message, table_name)
+
+
+def _coerce_dates(data: Iterable[dict]) -> List[dict]:
+    """
+    This is required to get dates into a string format that will be correctly picked up by the connector wire protocol.
+    :param data:
+    :return:
+    """
+    data_copy = []
+    for row in data:
+        row_copy = {}
+        for col, val in row.items():
+            if type(val) == date:
+                row_copy[col] = datetime.combine(val, time())
```

Comment:
the combine is neat, why do you use `datetime.time()` instead of `datatime.datetime.now()`?

Reply:
No reason, what would that give me?

Reply:
you're using datetime because sqlalchemy has nice integrations with datetime objects i think, but you're constructing the dates from time library -> datetime object. just a nit, not a big deal